### PR TITLE
[JFG-969] Java test configuration builders with mandatory and optiona…

### PR DESCRIPTION
**Pull request by a-badaev**
…l parameters

### CODE REVIEW

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestGroup.java [hunk 1]
```diff
@@ -1,51 +1,36 @@
...
 public class JTestGroup {
-    private String id;
-    private List<JTest> tests;
+    private final String id;
+    private final List<JTest> tests;
 
-
-    public static Builder builder() {
-        return new Builder();
+    public static Builder builder(Id id, List<JTest> tests) {
```
Comment (L-Evg) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestGroup.java]: May we have a vararg version of the method? I.e. 
builder(Id id, JTest[]...jTests)

That will simplify code of jagger tests i.e. no need to use Arrays.asList().

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestSuite.java [hunk 2]
```diff
@@ -1,53 +1,36 @@
...
 public class JTestSuite {
 
-    private String id;
-    private List<JTestGroup> testGroups;
+    private final String id;
+    private final List<JTestGroup> testGroups;
 
-    public static Builder builder() {
-        return new Builder();
+    public static Builder builder(Id id, List<JTestGroup> testGroups) {
```
Comment (L-Evg) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestSuite.java]: Again, what about a vararg version of such builder methods?

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/MaxLoadThreads.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * Date: 16/11/16
+ */
+public final class MaxLoadThreads {
+    private final long maxLoadThreads;
+    
+    private MaxLoadThreads(long maxLoadThreads) {
+        if (maxLoadThreads <= 0) {
+            throw new IllegalArgumentException(
+                    String.format("The maximum number of threads must be >= 0. Provided value is %s", maxLoadThreads));
+        }
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/MaxLoadThreads.java]: Incorrect message. Shoul be >

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/RequestsPerSecond.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * Date: 16/11/16
+ */
+public final class RequestsPerSecond {
+    private final long requestsPerSecond;
+    
+    RequestsPerSecond(long requestsPerSecond) {
+        if (requestsPerSecond <= 0) {
+            throw new IllegalArgumentException(
+                    String.format("Requests per second must be >= 0. Provided value is %s", requestsPerSecond));
+        }
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/RequestsPerSecond.java]: Same

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/WarmUpTimeInSeconds.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ * Date: 16/11/16
+ */
+public final class WarmUpTimeInSeconds {
+    private final long warmUpTimeInSeconds;
+    
+    private WarmUpTimeInSeconds(long warmUpTimeInSeconds) {
+        if (warmUpTimeInSeconds < 0) {
+            throw new IllegalArgumentException(
+                    String.format("The warm up time value in seconds. must be > 0. Provided value is %s",
+                            warmUpTimeInSeconds
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/load/WarmUpTimeInSeconds.java]: Incorrect message. Shoul be >=
Please check everywhere

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/Id.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.griddynamics.jagger.user.test.configurations;
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/Id.java]: Propose to move all auxiliary classes to separate package
/configurations/aux or /configurations/helpers

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestGroup.java (hunk 1 above)
Comment (anton-antonenko) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestGroup.java]: Now it's possible to pass empty List

Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestGroup.java]: yes. but it will be intentionally not accidentally

Comment (oskliarov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestGroup.java]: may be we can merge next builder with this
 builder(Id id, JTest test, JTest... tests) ?

**Pull request merged**